docs/samples/topic/authentication.md

```diff
@@ -52,5 +153,163 @@ pack.setUserAuthentication({
+  getConnectionName: async function(context) {
+    // Retrieve the endpoint that the user set.
+    let endpoint = context.endpoint;
+    let url = endpoint + "/api/v1/org";
+    let response = await context.fetcher.fetch({
+      method: "GET",
+      url: url,
+    });
+    return response.body.companyName;
+  },
+});
+
+// Allow the pack to make requests to Okta.
+pack.addNetworkDomain("okta.com");
+```
+## Automatic endpoint
+Authentication that automatically determines the account-specific endpoint URL during the OAuth2 flow. This sample connects to the Salesforce API.
+
+```ts
+import * as coda from "@codahq/packs-sdk";
+export const pack = coda.newPack();
+
+// Per-user authentication to the Salesforce API, using OAuth2 and an
+// automatically determined account-specific endpoint.
+pack.setUserAuthentication({
+  type: coda.AuthenticationType.OAuth2,
+  authorizationUrl: "https://login.salesforce.com/services/oauth2/authorize",
+  tokenUrl: "https://login.salesforce.com/services/oauth2/token",
+  scopes: ["id", "api", "refresh_token"],
+  additionalParams: {
+    prompt: "consent",
+  },
+
+  // Each account is associated with an instance URL, returned in the OAuth
+  // token response.
+  requiresEndpointUrl: true,
+  endpointDomain: "salesforce.com",
+  endpointKey: "instance_url",
+
+  // Determines the display name of the connected account.
+  getConnectionName: async function (context) {
+    // Relative URLs have the endpoint URL automatically prepended.
+    let url = "/services/oauth2/userinfo";
+    let response = await context.fetcher.fetch({
+      method: "GET",
+      url: url,
+    });
+    return response.body.name;
+  },
+});
+
+// Allow the pack to make requests to Salesforce.
+pack.addNetworkDomain("salesforce.com");
+```
+## User-selected endpoint
+Authentication that presents a list of endpoints to the user for them to select one. This sample connects to the Jira API.
+
+```ts
+import * as coda from "@codahq/packs-sdk";
+export const pack = coda.newPack();
+
+// Per-user authentication to the Jira API, using OAuth2 with a post-submit step
+// to select the instance to connect to.
+pack.setUserAuthentication({
+  type: coda.AuthenticationType.OAuth2,
+  authorizationUrl: "https://auth.atlassian.com/authorize",
+  tokenUrl: "https://auth.atlassian.com/oauth/token",
+  scopes: ["offline_access", "read:jira-user", "read:jira-work"],
+  additionalParams: {
+    audience: "api.atlassian.com",
+    prompt: "consent",
+    response_type: "code",
+  },
+
+  // After approving access, the user should select which instance they want to
+  // connect to.
+  requiresEndpointUrl: true,
+  endpointDomain: "atlassian.com",
+  postSetup: [{
+    type: coda.PostSetupType.SetEndpoint,
+    name: "SelectEndpoint",
+    description: "Select the site to connect to:",
+    // Determine the list of sites they have access to.
+    getOptionsFormula: coda.makeMetadataFormula(async function (context) {
```

Comment:
Ditto here, let's remove `makeMetadataFormula` if possible, otherwise let's fix the underlying issue there.